### PR TITLE
DDF-3840 Remove dependency on email address for user accounts

### PR DESCRIPTION
**Pull request by troymohl**
#### What does this PR do?
Removes dependency on user accounts having email addresses.

#### Who is reviewing it? 
@GabrielFabian 
@middlets719 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams

#### Ask 2 committers to review/merge the PR and tag them here. If you don't know who to ask, you can request reviews in https://groups.google.com/forum/#!forum/ddf-developers .
(please choose ONLY two committers from below, delete the rest)
@jlcsmith 
@stustison

#### How should this be tested? (List steps with links to updated documentation)
Full build with tests.

#### Any background context you want to provide?
User accounts may not always have an associated email address. This removes the dependency on email address and also cleans up some error reporting in the UI.

#### What are the relevant tickets?
[DDF-3840](https://codice.atlassian.net/browse/DDF-3840)

#### Checklist:
- [ ] Documentation Updated
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/security/catalog-security-plugin/src/main/java/ddf/catalog/security/plugin/SecurityPlugin.java [hunk 1]
```diff
@@ -51,10 +52,10 @@ public CreateRequest processPreCreate(CreateRequest input) throws StopProcessing
           .getMetacards()
           .forEach(
               metacard -> {
-                if (metacard.getTags().isEmpty() || metacard.getTags().contains("resource")) {
-                  metacard.setAttribute(
-                      new AttributeImpl(
-                          Metacard.POINT_OF_CONTACT, SubjectUtils.getEmailAddress(subject)));
+                String emailAddress = SubjectUtils.getEmailAddress(subject);
```
Comment (stustison) [inline catalog/security/catalog-security-plugin/src/main/java/ddf/catalog/security/plugin/SecurityPlugin.java]: Is there a reason why we wouldn't want to use the unique identifier here as well?

**Comment by bdeining** [comment]
build now

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java [hunk 2]
```diff
@@ -387,9 +395,11 @@ public void init() {
     post(
         "/subscribe/:id",
         (req, res) -> {
+          String userid = getSubjectIdentifier();
           String email = getSubjectEmail();
```
Comment (SmithJosh) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java]: ❓ Can we use the unique identifier for workspace operations too? This `getSubjectEmail` method expects the user's email to have an attribute key of `http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress`, which may not be the case. If it were configurable (i.e. using the unique identifier), that wouldn't be a problem.

Comment (troymohl) [reply catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java]: The workspace subscription stuff is email only, so I didn't replace that logic here with the unique ID as that may not return email. Emails should be in `http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress` claim.

**Pull request merged**